fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -793,19 +793,7 @@ public void finishTransaction(long transactionId, Set<Long> errorReplicaIds) thr
             errorReplicaIds.addAll(originalErrorReplicas);
         }
 
-        Database db = catalog.getDbNullable(transactionState.getDbId());
-        if (db == null) {
-            writeLock();
-            try {
-                transactionState.setTransactionStatus(TransactionStatus.ABORTED);
-                transactionState.setReason("db is dropped");
-                LOG.warn("db is dropped during transaction, abort transaction {}", transactionState);
-                unprotectUpsertTransactionState(transactionState, false);
-                return;
-            } finally {
-                writeUnlock();
-            }
-        }
+        Database db = catalog.getDbOrMetaException(transactionState.getDbId());
```

Comment:
Why not abort transaction when db is droped? Because it maybe recovered?

Reply:
committed txn will be checked when db to be dropped. it will be aborted outside